ice_fishing_abm_1/ice_fishing_abm_gp/tests/test_movement_destination_subroutine.py

```diff
@@ -0,0 +1,98 @@
+import pytest
+import numpy as np
+from pytest import fixture
+from ice_fishing_abm_1.ice_fishing_abm_gp.movement_destination_subroutine import ExplorationStrategy, GPExplorationStrategy
+
+@fixture
+def exploration_strategy():
+    return ExplorationStrategy()
+
+@fixture
+def gp_exploration_strategy():
+    return GPExplorationStrategy()
+
+def test_initialization_exploration_strategy(exploration_strategy):
+    assert exploration_strategy.grid_size == 100
+    assert exploration_strategy.ucb_beta == 0.2
+    assert exploration_strategy.softmax_tau == 0.01
+    assert exploration_strategy.mesh.shape == (10000, 2)
+    assert exploration_strategy.belief_softmax.shape == (100, 100)
+
+def test_initialization_gp_exploration_strategy(gp_exploration_strategy):
+    assert gp_exploration_strategy.grid_size == 100
+    assert gp_exploration_strategy.w_social == 0.4
+    assert gp_exploration_strategy.w_success == 0.3
+    assert gp_exploration_strategy.w_failure == 0.3
+    assert gp_exploration_strategy.social_feature_m.shape == (100, 100)
+    assert gp_exploration_strategy.success_feature_std.shape == (100, 100)
+
+def test_calculate_features_with_no_locations(gp_exploration_strategy):
```

Comment:
Since initializing the model is not costly, I don't think it's justified to use fixture here, just initializing GPExplorationStrategy here should be fine